shotover-proxy/src/transforms/cassandra/sink_cluster/mod.rs

```diff
@@ -515,6 +518,21 @@ impl CassandraSinkCluster {
         Ok(())
     }
 
+    async fn recreate_control_connection_from_node_in_pool(&mut self) -> Result<()> {
+        let random_address = self
```

Comment:
Not a random address, also could we pull out this logic into a common `create_control_connection` function and use it for creating the initial one too?

Reply:
Good point, I've refactored accordingly.